packages/api/src/router/post.ts

```diff
@@ -1,27 +1,72 @@
+import { TRPCError } from "@trpc/server";
 import { z } from "zod";
 
-import { createTRPCRouter, publicProcedure } from "../trpc";
+import { createTRPCRouter, protectedProcedure, publicProcedure } from "../trpc";
 
 export const postRouter = createTRPCRouter({
   all: publicProcedure.query(({ ctx }) => {
-    return ctx.prisma.post.findMany({ orderBy: { id: "desc" } });
+    return ctx.prisma.post.findMany({
+      include: { author: { select: { name: true, image: true } } },
+      orderBy: { id: "desc" },
+    });
   }),
+
   byId: publicProcedure
     .input(z.object({ id: z.string() }))
     .query(({ ctx, input }) => {
-      return ctx.prisma.post.findFirst({ where: { id: input.id } });
+      return ctx.prisma.post.findFirst({
+        include: { author: { select: { name: true, image: true } } },
+        where: { id: input.id },
+      });
     }),
-  create: publicProcedure
+
+  create: protectedProcedure
     .input(
       z.object({
         title: z.string().min(1),
         content: z.string().min(1),
       }),
     )
-    .mutation(({ ctx, input }) => {
-      return ctx.prisma.post.create({ data: input });
+    .mutation(async ({ ctx, input }) => {
+      function getNameFromUser() {
+        const meta = ctx.user.user_metadata;
+        if (typeof meta.name === "string") return meta.name;
+        if (typeof meta.full_name === "string") return meta.full_name;
+        if (typeof meta.user_name === "string") return meta.user_name;
+        return "[redacted]";
+      }
```

Comment:
is this the best way 😨 - it's not terrible but it's not the prettiest :)

Reply:
There is a way to use database triggers to copy over what we have in the metadata to public.profiles table
https://supabase.com/docs/guides/auth/managing-user-data#advanced-techniques